core/object.go

```diff
@@ -395,7 +401,8 @@ func init() {
 		Nil:           regType("Nil", (*Nil)(nil), "The 'nil' value"),
 		NodeSeq:       RegRefType("NodeSeq", (*NodeSeq)(nil), ""),
 		ParseError:    RegRefType("ParseError", (*ParseError)(nil), ""),
-		Proc:          RegRefType("Proc", (*Proc)(nil), "A callable function implemented via Go code"),
+		Proc:          RegRefType("Proc", (*Proc)(nil), "A self-identifying callable function implemented via Go code"),
+		ProcFn:        RegRefType("ProcFn", (*ProcFn)(nil), "A callable function implemented via Go code"),
```

Comment:
Does `ProcFn` really need to be exposed as Joker type?

Reply:
> Does `ProcFn` really need to be exposed as Joker type?

I'm not sure. Originally it was needed because only the `core` procedures were converted over; I converted the `std` procedures over much later during this project.

The only unwrapped `ProcFn` of which I'm unaware is the lazy-seq-supporting one in `csv_native.go`. Offhand, I'm not sure whether that'd ever be exposed; I haven't investigated Joker's implementations of lazy sequences.

I'm okay with removing it; let me know if you want me to do that.

Reply:
Yes, let's remove it. We can always add it back if needed.

Reply:
Done!